src/VirtoCommerce.ImageToolsModule.Data/ThumbnailGeneration/ThumbnailGenerationProcessor.cs

```diff
@@ -16,6 +17,8 @@ public class ThumbnailGenerationProcessor : IThumbnailGenerationProcessor
         private readonly ISettingsManager _settingsManager;
         private readonly IImagesChangesProvider _imageChangesProvider;
 
+        private readonly HashSet<string> tasksToExpireChangesCache = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Seems this HashSet member is redundant, you can do the same logic without using an extra member
using only `ProcessTasksAsync` method arguments.